client/.eslintrc.js

```diff
@@ -147,6 +147,13 @@ module.exports = {
     'object-curly-newline': ['error', { consistent: true }],
     'prefer-destructuring': 'off',
     'no-restricted-exports': 'off',
+    'no-param-reassign': [
```

Comment:
Can we do the same whitelisting for TypeScript below too? Currently, it's just set to `off`, which I think is also rather unnerving.

Reply:
As discussed, this will trigger error for `produce` and `reduce` usage. We keep this as of now.